### PR TITLE
Day 5: React - Introduction to State

### PR DESCRIPTION
**Pull request by AnimeshKumarSuccessive**
Animesh Kumar
Ticket number: [40431](https://redmine.successive.tech/issues/40431)

### CODE REVIEW

File: src/components/Slider/Slider.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  } = props;
+
+  const [index, setIndex] = useState(0);
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      if (random) {
+        setIndex(getRandomNumber(MAX_NUMBER));
+      } else {
+        setIndex((i) => getNextRoundRobin(MAX_NUMBER, i));
```
Comment (yogeshsinghh) [inline src/components/Slider/Slider.jsx]: Use banners length as the max number

File: src/components/Slider/Slider.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      }
+    },
+    duration);
+    return () => clearInterval(interval);
+  }, []);
+
+  return (
+    <>
+      <div style={sliderContainer}>
+        <img src={banners[index] ? banners[index] : defaultBanner} alt={altText} height={height} />
```
Comment (yogeshsinghh) [inline src/components/Slider/Slider.jsx:29]: import Public folder path value and use for the image src path with the banners[index]

File: src/components/Slider/Slider.jsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  );
+};
+
+Slider.propTypes = {
+  altText: PropTypes.string.isRequired,
+  banners: PropTypes.arrayOf.isRequired,
+  height: PropTypes.number.isRequired,
+  duration: PropTypes.number.isRequired,
+  random: PropTypes.number.isRequired,
+  defaultBanner: PropTypes.string.isRequired,
```
Comment (yogeshsinghh) [inline src/components/Slider/Slider.jsx]: Use dafault values of the props

File: src/pages/TextFieldDemo/TextFieldDemo.jsx [hunk 4]
```diff
@@ -1,16 +1,31 @@
...
-  </div>
-);
+const TextFieldDemo = () => {
+  const images = [
+    `${PUBLIC_IMAGE_FOLDER}cloud.jpg`,
+    `${PUBLIC_IMAGE_FOLDER}dns-server.png`,
+    `${PUBLIC_IMAGE_FOLDER}full-stack-web-development.jpg`,
+    `${PUBLIC_IMAGE_FOLDER}js.jpg`,
+    `${PUBLIC_IMAGE_FOLDER}load-balancer.png`,
+  ];
```
Comment (yogeshsinghh) [inline src/pages/TextFieldDemo/TextFieldDemo.jsx]: move this array to constant file  as array of name of images  ex ["cloud.jpg", ....]

File: src/pages/TextFieldDemo/TextFieldDemo.jsx [hunk 5]
```diff
@@ -1,16 +1,31 @@
...
 
-export default TexFieldDemo;
+  return (
+    <>
+      <div style={mainContainer}>
+        <Slider altText="Image not found" banners={images} defaultBanner={DEFAULT_BANNER_IMAGE} duration={2000} height={200} random={5} />
```
Comment (yogeshsinghh) [inline src/pages/TextFieldDemo/TextFieldDemo.jsx]: send random as the boolean

**Review by yogeshsinghh — CHANGES_REQUESTED** [review]

File: src/pages/TextFieldDemo/TextFieldDemo.jsx (hunk 5 above)
Comment (AnimeshKumarSuccessive) [reply src/pages/TextFieldDemo/TextFieldDemo.jsx]: Done

File: src/components/Slider/Slider.jsx (hunk 3 above)
Comment (AnimeshKumarSuccessive) [reply src/components/Slider/Slider.jsx]: Done

File: src/pages/TextFieldDemo/TextFieldDemo.jsx (hunk 4 above)
Comment (AnimeshKumarSuccessive) [reply src/pages/TextFieldDemo/TextFieldDemo.jsx]: Done

File: src/components/Slider/Slider.jsx (hunk 1 above)
Comment (AnimeshKumarSuccessive) [reply src/components/Slider/Slider.jsx]: Done

**Review by yogeshsinghh — APPROVED** [review]

**Pull request merged**